x/gamm/types/pool.go

```diff
@@ -141,6 +137,16 @@ type WeightedPoolExtension interface {
 	GetTokenWeight(denom string) (sdk.Int, error)
 }
 
+type BalancerPool interface {
+	PoolI
+	TraditionalAmmInterface
+}
+
+type StableSwapExtension interface {
+	PoolI
+	TraditionalAmmInterface
+}
+
```

Comment:
Note to reviewer, need to decide what we are adding to these extensions. Was not clear to me what is not shared since I did not work on stable swap

Reply:
Most logic is shared and we shouldn't need these.

Balancer pool supports more methods but those are defined on other / additional interface extensions such as `PoolAmountOutExtension`